test/algorithms/test_partial_equivalence.cpp

```diff
@@ -0,0 +1,420 @@
+      controlledSwapGate,
+      dd->multiply(hGate2, dd->multiply(zGate2, controlledSwapGate)));
+  const auto c4 = dd->multiply(controlledHGate2, xGate);
+
+  EXPECT_FALSE(dd->zeroAncillaePartialEquivalenceCheck(c3, c4, 1));
+}
+
+TEST(PartialEquivalenceTest, DifferentNumberOfQubits) {
+  const auto dd = std::make_unique<dd::Package<>>(3);
+  const auto controlledSwapGate =
+      dd->makeTwoQubitGateDD(dd::SWAP_MAT, 3, qc::Controls{1}, 0, 2);
+  const auto hGate = dd->makeGateDD(dd::H_MAT, 3, 0);
+  const auto zGate = dd->makeGateDD(dd::Z_MAT, 3, 2);
+  const auto xGate = dd->makeGateDD(dd::X_MAT, 2, 1);
+  const auto controlledHGate = dd->makeGateDD(dd::H_MAT, 2, qc::Controls{1}, 0);
+
+  const auto c1 = dd->multiply(
+      controlledSwapGate,
+      dd->multiply(hGate, dd->multiply(zGate, controlledSwapGate)));
+  const auto c2 = dd->multiply(controlledHGate, xGate);
+
+  EXPECT_TRUE(dd->partialEquivalenceCheck(c1, c2, 3, 1));
+  EXPECT_FALSE(dd->partialEquivalenceCheck(c2, c1, 3, 3));
+  EXPECT_FALSE(dd->partialEquivalenceCheck(c2, dd::mEdge::zero(), 2, 1));
+  EXPECT_FALSE(dd->partialEquivalenceCheck(c2, dd::mEdge::one(), 2, 1));
+  EXPECT_FALSE(dd->partialEquivalenceCheck(dd::mEdge::one(), c1, 2, 1));
+  EXPECT_TRUE(
+      dd->partialEquivalenceCheck(dd::mEdge::one(), dd::mEdge::one(), 0, 1));
+  EXPECT_TRUE(
+      dd->partialEquivalenceCheck(dd::mEdge::one(), dd::mEdge::one(), 0, 0));
+}
+
+TEST(PartialEquivalenceTest, ComputeTableTest) {
+  const auto nqubits = 3U;
+  const auto dd = std::make_unique<dd::Package<>>(nqubits);
+  const auto controlledSwapGate =
+      dd->makeTwoQubitGateDD(dd::SWAP_MAT, nqubits, qc::Controls{1}, 2, 0);
+  const auto hGate = dd->makeGateDD(dd::H_MAT, nqubits, 0);
+  const auto zGate = dd->makeGateDD(dd::Z_MAT, nqubits, 2);
+  const auto xGate = dd->makeGateDD(dd::X_MAT, nqubits, 1);
+  const auto controlledHGate =
+      dd->makeGateDD(dd::H_MAT, nqubits, qc::Controls{1}, 0);
+
+  const auto c1 = dd->multiply(
+      controlledSwapGate,
+      dd->multiply(hGate, dd->multiply(zGate, controlledSwapGate)));
+  const auto c2 = dd->multiply(controlledHGate, xGate);
+
+  EXPECT_TRUE(dd->partialEquivalenceCheck(c1, c2, 3, 1));
+  EXPECT_TRUE(dd->partialEquivalenceCheck(c1, c2, 3, 1));
+  EXPECT_TRUE(dd->partialEquivalenceCheck(c1, c2, 3, 1));
+  EXPECT_TRUE(dd->partialEquivalenceCheck(c1, c2, 3, 1));
+  EXPECT_TRUE(dd->partialEquivalenceCheck(c1, c2, 3, 1));
+}
+
+TEST(PartialEquivalenceTest, MQTBenchGrover3Qubits) {
+  const auto dd = std::make_unique<dd::Package<>>(7);
+
+  const qc::QuantumComputation c1{
+      "./circuits/grover-noancilla_nativegates_ibm_qiskit_opt0_3.qasm"};
+  const qc::QuantumComputation c2{
+      "./circuits/grover-noancilla_indep_qiskit_3.qasm"};
+
+  // 3 measured qubits and 3 data qubits, full equivalence
+  EXPECT_TRUE(dd->partialEquivalenceCheck(
+      buildFunctionality(&c1, *dd, false, false),
+      buildFunctionality(&c2, *dd, false, false), 3, 3));
+}
```

Comment:
I think I would prefer to move all the tests that involve files to QCEC once the corresponding feature is added there.
It should be enough to test the basic functionality here and leave more elaborate tests/benchmarks to the top-level repository.
This also means, that you can remove the `circuits` directory.

Reply:
I assume that you mean that I should remove the circuits from the directory, not remove the whole directory, as it was already there beforehand 🤔

Reply:
Exactly 😌